pkg/p2p/streamtest/streamtest.go

```diff
@@ -147,12 +155,42 @@ func (s *stream) Write(p []byte) (int, error) {
 }
 
 func (s *stream) Close() error {
+	// lock + closed is used to avoid multiple closing or sending on a closed channel
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if s.closed {
+		return nil
+	}
+
 	if err := s.in.Close(); err != nil {
 		return err
 	}
 	if err := s.out.Close(); err != nil {
 		return err
 	}
+
+	select {
+	case s.cin <- struct{}{}:
+		close(s.cin)
+		s.closed = true
+	default:
+	}
+
+	return nil
+}
+
+func (s *stream) FullClose() error {
+	if err := s.Close(); err != nil {
+		return err
+	}
+
+	select {
+	case <-s.cout:
+	case <-time.After(1 * time.Second):
```

Comment:
Isn't this time too short?

Reply:
I thought to actually export it in streamtest so it can be configured, maybe leave a default to 1sec. I wanted to avoid waiting for longer time in tests where we expect timeout.

Reply:
Then it would be good to make a global variable that is updated (maybe with some convenient function in export_test.go) in every test only and reset after the each test is done.

Reply:
Done, I am not sure if that is what you had in mind though :)